### ISSUE TITLE
[Proposal] Configuration files for embedded version

### ISSUE THREAD

**Issue by aaaaaaaaargh** [osxmidi/LinVst#68]
I think it would be a nice addition to support per-application (resp. per-directory) configuration files to support settings like wine prefixes that otherwise would be heavily dependent on the VST host that is utilizing LinVst. My proposal would be to have the dll / linvst host look for a config file, namely .linvst.conf or something like that, and traverse towards the filesystem root until such a file has been found (or not).

I consider myself an experienced C++ developer so guess I could help with the implementation of such a thing. Please let me know how you think about it.

**Comment by osxmidi** [comment]
Yes it seems like a good idea and I have previously thought about it.

As it stands now, if the (renamed) linvst.so file is located within some wineprefix and is then loaded by the DAW then it (the vst plugin) will pick up that particular individual wineprefix settings automatically (that wineprefixes dll overrides etc).

If you come up with anything about implementing a config file then I'm open to discussing it.

Maybe what it does now with wineprefixes plus the ability to use a config file would be the best way.

**Comment by aaaaaaaaargh** [comment]
Oh is that true? I wasn't aware the current wine prefix is already being picked up. However, I am currently faced with a scenario where your VSTs are actually not inside a wineprefix but rather, lets say, in ~/Audio/VSTs while the wineprefix is somewhere else. I'm not sure whether this is a common scenario or I'm just a weird person for wanting it like that (I got my good reasons for that though).

It would be nice to have the linvst.so wrapper file (i mean the renamed one) scan from its position down to the fs root for some fixed named config file. I did this a couple of times in other projects and it turned out to be a nice pattern as you basically get inheritance for granted - if wanted.

I'm supporting your suggestion of having both explicit (config file) and implicit (current wineprefix) methods, because it wouldn't break the current way things work. Honestly I did not look at the code yet (with the exception of the convert tools), but I guess it shouldn't be that hard to implement such a feature.
